numpy/random/tests/test_randomstate.py

```diff
@@ -107,6 +107,23 @@ def test_cannot_seed(self):
     def test_invalid_initialization(self):
         assert_raises(ValueError, random.RandomState, MT19937)
 
+    def test_init_unique(self):
+        # gh-14844. seed[0] was not correctly initialized;
+        # key[0] always held the same value
+        rs = random.RandomState()
+        key = rs._bit_generator.state['state']['key']
+        res = np.empty((32,) + key.shape, dtype=key.dtype)
+        for i in range(32):
+            # enter the _legacy_seed(None) path
+            rs.seed()
+            key = rs._bit_generator.state['state']['key']
+            res[i, ...] = key
+        # make sure each the corresponding values at each position in key
+        # are unique. The chances of that failing should be very low.
+        res.shape = (32, -1)
+        for i in range(res.shape[1]):
+            assert len(np.unique(res[:, i])) == 32
```

Comment:
I might suggest a more targeted regression test for the actual bug: `assert not (res[:, 0] == [0x80000000]*32).all()`

Reply:
adopted